android/src/test/java/com/amplitude/android/AmplitudeTest.kt

```diff
@@ -119,6 +119,8 @@ class AmplitudeTest {
             event.eventType = "test event"
             amplitude?.track(event)
             advanceUntilIdle()
+            Thread.sleep(100)
```

Comment:
1. I can increment it to something like `500` but it will increase test duration.
2. I would use more stable ways (something like `???.await()`) but I am not sure what is better/possible to use in Kotlin and do not add notable memory/cpu overhead.

Reply:
I don't believe increasing sleep time is a good option. Not sure if @justin-fiedler or @bgiori know some better ways to mock this in tests in Kotlin though. If not, it maybe fine if you run it several times on local and this test always passes.

Reply:
yes, I ran the tests with `Thread.sleep` many times locally and they always pass